### ISSUE TITLE
Add time_limit option

### ISSUE THREAD

**Issue by ebarnard** [oxfordcontrol/osqp#27]
It is useful in soft real-time mpc to be able to have large optimisations run in the free time between mpc iterations. For this to work it must be possible to interrupt the solver at arbitrary times.

This is almost possible using check_termination with a value of 1 and the ctrl-c handler. However, this does not work on windows as signal handlers are per-process so this breaks with multiple osqp solvers running (and does not restore the current signal handler).

I'd suggest adding an interrupt callback that executes at the same time the ctrl-c handler and can be enabled irrespective of the value of the CTRLC build option using:

`c_int osqp_set_interrupt_callback(OSQPWorkspace* work, c_int (*callback)(void* opaque), void* opaque);`

The callback is disabled if a value of `NULL` is passed to `callback`.

I'm happy to implement this. Just want feedback.

**Comment by bstellato** [comment]
That's actually something I thought about. I think it should not be a problem to implement the callback as you described. However, I see that many other solvers deal with this issue with a setting to specify the maximum execution time. 

For example, GUROBI has [TimeLimit](http://www.gurobi.com/documentation/7.5/refman/timelimit.html). Also qpOASES, which is used in many real-time applications, has the `cputime` option. We could add this option (as a `cfloat` with value `0` if disabled) together with a specific return status in case the maximum time is reached. I think it would be easier than handling callbacks. Would it work for you?

**Comment by ebarnard** [comment]
For my use case a time limit would work great, so let's do a time_limit option.

A cross-platform timer implementation as a self-contained C file: https://github.com/nclack/tictoc/tree/master/src.

There is another issue that that it is not possible to interrupt solvers running in parallel on Windows and it is not possible to interrupt solvers without also making them nondeterministically interruptible via ctrl-c, if running in a terminal. But this probably deserves a separate issue.

**Comment by bstellato** [comment]
We have already implemented a timer in OSQP to return the setup and solution time [here](https://github.com/oxfordcontrol/osqp/blob/2aeaebb3a3f8f718e16373caccb0c5a8e091a004/include/util.h#L135).
I think it is just a matter of adding a termination check to the right places of the [osqp_solve](https://github.com/oxfordcontrol/osqp/blob/2aeaebb3a3f8f718e16373caccb0c5a8e091a004/src/osqp.c#L220) function. We can make a branch and merge it to the `develop` branch when it is ready.

If real-time execution is the main concern, I think the time_limit option should work. We can discuss the other issue separately.
